### ISSUE TITLE
[java] NonThreadSafeSingleton: False negative with property checkNonStaticMethods

### ISSUE THREAD

**Issue by JavaGuangGitHub** [pmd/pmd#6780]
Body:

**Affects PMD Version:** 7.25.0 (also reproducible in 6.44.0)

**Rule:** [NonThreadSafeSingleton](https://docs.pmd-code.org/latest/pmd_rules_java_multithreading.html#nonthreadsafesingleton)

**Description:**

The property `checkNonStaticMethods` has the description:

> Check for non-static methods. Do not set this to false and checkNonStaticFields to true.

The natural reading is that when `checkNonStaticMethods=true`, the rule **checks** non-static methods. However, the implementation does the opposite.

In `NonThreadSafeSingletonRule.java`:

```java
if (checkNonStaticMethods && !node.isStatic()
        || node.isSynchronized()) {
    return data; // skip the method
}
```

When checkNonStaticMethods=true (the default), non-static methods are skipped, not checked. Conversely, setting it to false causes non-static methods to be analyzed.

This is inconsistent with the property name and description.

Minimal reproduction:

```java
public class Singleton {
    private static Singleton instance;

    // Non-static getInstance - should be flagged by default
    public Singleton getInstance() {
        if (instance == null) {
            instance = new Singleton();
        }
        return instance;
    }
}
```

With default properties (checkNonStaticMethods=true), PMD reports 0 violations.
Setting checkNonStaticMethods=false reports 1 violation.

Expected behavior:

The behavior should match the description, i.e.:

checkNonStaticMethods=true (default): non-static methods are checked.
checkNonStaticMethods=false: non-static methods are skipped.
Alternatively, if the current behavior is intentional, the property name/description should be changed to something like skipNonStaticMethods to avoid confusion.

Running PMD via: any (verified with unit tests against pmd-java 6.44.0 and 7.25.0)
